client/cli/src/lib.rs

```diff
@@ -388,7 +388,25 @@ impl<'a> ParseAndPrepareExport<'a> {
 			None => Box::new(stdout()),
 		};
 
-		builder(config)?.export_blocks(exit.into_exit(), file, from.into(), to.map(Into::into), json)?;
+		// Note: while we would like the user to handle the exit themselves, we handle it here
+		// for backwards compatibility reasons.
```

Comment:
What kind of backwards compatibility? To what?

Reply:
To keep the API of `substrate-cli` backwards-compatible.
"We would like the user [of substrate-cli] to handle the exit"

Reply:
And when do you plan to let the user handle this?

Reply:
If we let the user handle this, then we'd need to propagate the changes to Polkadot, which in turns would require a refactoring of `Worker` trait first, which itself is more complicated than it looks because it requires passing a `&Service`.